File: vera-shawn/who-likes-it.js
// BEGIN
// INIT - variables and functions
// implement a function that takes an array and returns formatted text
// if 4 or more names are entered the number in the "and 2 others" increases
function likes(names) {
    // TODO
    let likes = names.length;
   
    // if statements to determine output
    if (likes === 0) {
      return "no one likes this"
    } else if (likes = 1) {
      return likes;
    } else if (likes = 2) {
        return likes;
      } else if (likes = 3) {
          return likes + "1 other likes this"
        } else if (likes => 4) {
      return likes + "2 others like this"
    } 
     
    //END
  }

File: vera-shawn/who-likes-it.py
# BEGIN
# INPUT - an array of names of people that like an item
# IF - array is empty RETURN "no one likes this"
# ELIF 2 or 3 ppl like it list both names "so, so and so like this"
# ELIF if more than 3 like it RETURN "so, so and # others like this"
# END

def likes(names):
    
    array = len(names)
    
    if array == 0:
        return "no one likes this"
    elif array == 1:
        return names[0] + ' likes this'
    elif array == 2:
        return names[0] + ' and ' + names[1] + ' like this'
    elif array == 3:
        return names[0] + ', ' + names[1] + ' and ' + names[2] + ' like this'
    else:
        return names[0] + ', ' + names[1] + ' and ' + str(array - 2) + ' others like this'